src/cloudant/_common_util.py

```diff
@@ -288,3 +289,35 @@ class _Code(str):
     """
     def __new__(cls, code):
         return str.__new__(cls, code)
+
+class InfiniteSession(Session):
+    """
+    This class provides for the ability to automatically renew session login
+    information in the event of expired session authentication.
+    """
+
+    def __init__(self, username, password, server_url):
+        super(InfiniteSession, self).__init__()
+        self._username = username
+        self._password = password
+        self._server_url = server_url
+
+    def request(self, method, url, **kwargs):
+        """
+        Overrides ``requests.Session.request`` to perform a POST to the
+        _session endpoint to renew Session cookie authentication settings and
+        then retry the original request, if necessary.
+        """
+        resp = super(InfiniteSession, self).request(method, url, **kwargs)
+        path = url_parse(url).path.lower()
+        post_to_session = method.upper() == 'POST' and path == '/_session'
+        if resp.status_code == 401 and not post_to_session:
```

Comment:
Sure, makes sense.  Done in https://github.com/cloudant/python-cloudant/pull/239/commits/e88a5eb701421fcde605895f6a43f50710cbf765.